Fixed mobs attributes on spawn, added fiery mobs tag in place of Blaze mob

SojaIntegral opened the pull request:
Fixed attributes not being properly applied to entities on spawn (Dead king and Tyros) by changing FinalizeSpawnEvent to EntityJoinLevelEvent.

Also, added FieryMobs tag as was under a TODO comment.

### Contributor Liscence Agreement
[Full CLA](https://github.com/iron431/Irons-Spells-n-Spellbooks/blob/1.19.2/CLA.md)
*Contributors: Please "X" in the following box to acknowledge our CLA*
- [X] I have read and agree to the CLA for Iron's Spells and Spellbooks which allows me to retain my ownership in the code submitted while granting the repository owner legal rights to use my contribution.

SojaIntegral committed:
Fixed mobs attributes on spawn, added fiery mobs tag in place of Blaze mob (includes Tyros and Pyromancer)

iron431 reviewed (changes requested):
Thanks for taking initiative and helping out with some todos! However, there a few things that should be addressed before it can be merged in:

> EntityJoinLevelEvent makes sure the entity always has the attribute applied, even in already spawned, old entities

This looks problematic. We specifically don't want to apply the attributes multiple times, because it can conflict with mods or datapack that modify an entity's base attribute during its lifetime, and expect that change to stick. All mobs spawns should route themselves through the finalize spawn event, I believe finalize spawn is still the correct event to use here.

Also just to clarify, the idea is that only "fire elemental" mobs, mobs made of fire, should have the weakness, not just any mob that also has fire resistance.

Finally as a sidenote, feel free to remove the "meta" comments, that speech should be saved for on github and not embedded into the cobebase.

SojaIntegral committed:
Reviewed handleResistanceAttributesOnSpawn

Changed the system to apply a modifier instead of setting the value. The system also checks if the mob already has the modifier before applying, which should make so it never applies multiple times.
Also, this system should be completely compatible with any add-ons or mods that also modify attributes.

Reasoning:
FinalizeSpawnEvent does not work for DeadKing, Tyros, or many other mods that might benefit indirectly from the system, since it uses UNDEAD and SENSITIVE_TO_IMPALING tags, as well as fireImmune() for applying resistances.

EntityJoinLevelEvent ensures that any entity that fits the criteria will have their attributes modified and addUniqueModifier applies a modifier, not changing any work done by other mods, while also ensuring that the modifier is only applied once (avoiding attributes going crazy high).

SojaIntegral committed:
Changed to include only Fiery Mobs

I didn't understand it very well before, since it was right below one that included Pyromancer and others, but I think I get it now that you clarified it.

Changed the tag to only include Blaze and Magma Cube.

SojaIntegral commented:
The issue with using finalize spawn is that mobs like the Dead King, Tyros and some other modded entities from other mods don't get the attribute bonus (I guess it doesn't count as spawn?), hence why I used the join level event.

I made some changes, trying to address your concerns with my changes. I hope it satisfies you. If not, at least I tried and the tag I made wasn't that far off.
